packages/consumption/src/modules/requests/itemProcessors/readAttribute/ReadAttributeRequestItemProcessor.ts

```diff
@@ -1,49 +1,172 @@
+            attribute = foundLocalAttribute.content;
+
+            if (_requestItem.query instanceof IdentityAttributeQuery && attribute instanceof IdentityAttribute && this.accountController.identity.isMe(attribute.owner)) {
+                if (foundLocalAttribute.isShared()) {
+                    return ValidationResult.error(
+                        CoreErrors.requests.attributeQueryMismatch(
+                            "The provided IdentityAttribute is a shared copy of a RepositoryAttribute. You can only share RepositoryAttributes."
+                        )
+                    );
+                }
+
+                const ownSharedIdentityAttributeVersions = await this.consumptionController.attributes.getSharedVersionsOfRepositoryAttribute(
+                    foundLocalAttribute.id,
+                    [requestInfo.peer],
+                    false
+                );
+                const sourceAttributeIdsOfOwnSharedIdentityAttributeVersions = ownSharedIdentityAttributeVersions.map((ownSharedIdentityAttribute) =>
+                    ownSharedIdentityAttribute.shareInfo?.sourceAttribute?.toString()
+                );
+
+                let repositoryAttribute = foundLocalAttribute;
+                let i = 0;
+                while (repositoryAttribute.succeededBy !== undefined && i < 1000) {
+                    const successor = await this.consumptionController.attributes.getLocalAttribute(repositoryAttribute.succeededBy);
+                    if (!successor) {
+                        throw TransportCoreErrors.general.recordNotFound(LocalAttribute, repositoryAttribute.succeededBy.toString());
+                    }
+                    if (sourceAttributeIdsOfOwnSharedIdentityAttributeVersions.includes(successor.id.toString())) {
+                        return ValidationResult.error(
+                            CoreErrors.requests.attributeQueryMismatch(
+                                `The provided IdentityAttribute is outdated. You have already shared the Successor '${successor.id.toString()}' of it.`
+                            )
+                        );
+                    }
+                    repositoryAttribute = successor;
+                    i++;
+                }
+            }
+
+            if (_requestItem.query instanceof ThirdPartyRelationshipAttributeQuery && attribute instanceof RelationshipAttribute) {
+                if (!foundLocalAttribute.isShared()) {
+                    throw new Error("this should never happen");
+                }
+
+                if (foundLocalAttribute.shareInfo.sourceAttribute !== undefined) {
+                    return ValidationResult.error(
+                        CoreErrors.requests.attributeQueryMismatch(
+                            "When responding to a ThirdPartyRelationshipAttributeQuery, only RelationshipAttributes that are not a copy of a sourceAttribute may be provided."
```

Comment:
I think it is less error-prone in the context of Attribute deletion and other processes. An Attribute without `sourceAttribute` can be the `sourceAttribute` for several Attributes, but Attributes with `sourceAttribute` cannot be the `sourceAttribute`of another Attribute. If this were possible, longer chains of Attributes would be created with regard to sharing, which could make things more complicated. As it is not a restriction in terms of functionality to only allow the sharing of Attributes without `sourceAttribute`, I think this would be equally preferable for IdentityAttributes and RelationshipAttributes.